lib/IRGen/GenClass.cpp

```diff
@@ -451,28 +460,36 @@ namespace {
   };
 } // end anonymous namespace
 
-void ClassTypeInfo::generateLayout(IRGenModule &IGM, SILType classType) const {
-  assert(!FieldLayout && "already generated layout");
-
-  // Perform fragile layout if Objective-C interop is enabled.
-  bool CompletelyFragileLayout =
-      (IGM.Context.LangOpts.EnableObjCInterop &&
-       !IGM.IRGen.Opts.EnableClassResilience);
+void ClassTypeInfo::generateLayout(IRGenModule &IGM, SILType classType,
+                                   bool completelyFragileLayout) const {
+  unsigned idx = (completelyFragileLayout ? 1 : 0);
+  assert(!FieldLayout[idx] && "already generated layout");
 
   // Add the heap header.
   ClassLayoutBuilder builder(IGM, classType, Refcount,
-                             CompletelyFragileLayout);
+                             completelyFragileLayout);
 
   // generateLayout should not call itself recursively.
-  assert(!FieldLayout && "already generated layout re-entrantly");
+  assert(!FieldLayout[idx] && "already generated layout re-entrantly");
+
+  auto *classTy =
+      cast<llvm::StructType>(getStorageType()->getPointerElementType());
+
+  if (completelyFragileLayout) {
+    // Create a name for the new llvm type.
+    std::string typeName;
+    llvm::raw_string_ostream os(typeName);
+    os << classTy->getName() << "_fragile";
```

Comment:
Nitpick: SmallString / raw_svector_ostream. Or even just SmallString and +=, which is ever-so-slightly more optimizable.

Reply:
Ok. I copy and pasted this from the 'tail layout' code, so I'll fix that too.